console/views.py

```diff
@@ -111,14 +116,70 @@ def get_filtered_responses(tx_id, ru_ref, survey_id, datetime_earliest, datetime
     return filtered_data
 
 
-@app.route('/store', methods=['GET'])
+def encrypt_data(unencrypted_json):
+    logger.info('Encrypting data')
+    encrypter = Encrypter()
+    encrypted_data = encrypter.encrypt(unencrypted_json)
+    logger.info('Data successfully encrypted')
+
+    return encrypted_data
+
+
+def get_publisher(logger):
+    urls = settings.RABBIT_URLS
+    queue = settings.RABBIT_SURVEY_QUEUE
+    collect_publisher = QueuePublisher(logger, urls, queue)
+
+    return collect_publisher
+
+
+def publish_result(publisher, json_string):
+    tx_id = str(uuid.uuid4())
+    logger.info('Created new tx_id ' + tx_id)
+    json_string['tx_id'] = tx_id
+    json_string['survey_id'] = str(json_string['survey_id'])
+    encrypted_data = encrypt_data(json_string)
+
+    publisher.publish_message(encrypted_data, headers={'tx_id': tx_id})
+
+
+@app.route('/store', methods=['GET', 'POST'])
 @flask_security.roles_required('SDX-Developer')
 def store():
-    tx_id = request.args.get('tx_id', type=str, default='')
-    ru_ref = request.args.get('ru_ref', type=str, default='')
-    survey_id = request.args.get('survey_id', type=str, default='')
-    datetime_earliest = request.args.get('datetime_earliest', type=str, default='')
-    datetime_latest = request.args.get('datetime_latest', type=str, default='')
-
-    store_data = get_filtered_responses(tx_id, ru_ref, survey_id, datetime_earliest, datetime_latest)
-    return render_template('store.html', data=store_data)
+    if request.method == 'POST':
+        json_string = request.form['json_data']
```

Comment:
Do a `.get(` instead of square bracket notation to return none - just in case this key isn't available for whatever reason